src/_pytest/_code/code.py

```diff
@@ -213,7 +213,7 @@ def statement(self) -> "Source":
         return source.getstatement(self.lineno)
 
     @property
-    def path(self):
+    def path(self) -> Union[py.path.local, str]:
```

Comment:
`Code.path` has this annoying type due to some error path, we should fix it sometime...